rtl/eclrtl/eclrtl.cpp

```diff
@@ -4851,12 +4851,11 @@ class CStrRegExprFindInstance : implements IStrRegExprFindInstance
         {
             if (_keep)
             {
-                sample = (char *)rtlMalloc(_len + 1);  //required for findstr
+                sample = (char *)rtlMalloc(_len + 1);
                 memcpy(sample, _str + _from, _len);
-                sample[_len] = (char)NULL;
```

Comment:
This line has been lost.  If it is removed then the search call in the following line needs to be changed.